src/cml.js

```diff
@@ -160,19 +161,30 @@ class CML {
     throw new Error(`driver ${driver} unknown!`);
   }
 
+  watermark(driver, label = '') {
+    // Replace {workflow} and {run} placeholders in label with actual values.
+    const workflow = driver.workflowId;
+    const run = driver.runId;
+    label = label.replace('{workflow}', workflow);
+    label = label.replace('{run}', run);
+    label = `CML watermark ${label}`.trim();
+    return `![](${WATERMARK_IMAGE} "${label}")`;
+  }
```

Comment:
Makes no a big sense to make this function as a member of the class and pass the driver. It can retrieve the driver at any point. Also the signature of the functions is always passing an object as argument vs arguments. This way we then do named parameters calls.

I suggest to remove this function and make it within the specific function

```suggestion
```

Reply:
I understand adhering to a single function signature across the codebase. But I don't see how it's logical to merge the code for watermark creation into the `commentCreate` function's body. That function (`commentCreate`) is already over 150 lines long, my instinct is to remove code from it, not add more.

Reply:
I saw it, I would suggest a refactor later on replacing section of code within functions.
That way despite the method is still large it can be collapsed logically.
Reasons to not do the watermark method:

 - There is not point since its not very useful interfacing outside CML. The method in js is public (not private) and mangles a bit the interface
 
In any case the strangest thing is the driver passed as a parameter. It can get it at any point.
You can externalise the function also within the module like many others

```js
const genWatermark = ({ runId, workflowId, rmWatermark, watermarkTitle } = {}) => {
    if (rmWatermark) return '';
    const lbl = watermarkTitle.replace('{workflow}', workflowId).replace('{run}', runId).trim();
    return `![](${WATERMARK_IMAGE} "CML watermark ${lbl}")`;
  }
```

then call genWatermark

Reply:
In any case if we were encapsulating code within functions inside the member would be fine and should not be counted as X lines of code

Reply:
I have opened https://github.com/iterative/cml/issues/1234 @tasdomas

Reply:
Sure, I'll refactor.

But not counting embedded functions towards the line count seems like a bad way to move towards a readable and understandable codebase.

Reply:
lets see in the refactor... I have also the feeling that we are making it a bit complex in general